python/lsst/ip/isr/transmissionCurve.py

```diff
@@ -0,0 +1,214 @@
+
+        metadata = tableList[0].meta
+        calib.setMetadata(metadata)
+        calib.updateMetadata()
+
+        calib.data = tableList[0]
+        calib.setTransmissionCurveRepresentation()
+        return calib
+
+    def setTransmissionCurveRepresentation(self):
+        """Construct transmission curve representation from the data that was
+        read.
+
+        Raises
+        ------
+        RuntimeError
+            This is raised if no table data exists in the calibration,
+            if there are array length mismatches, or if the wavelength
+            sampling for multi-amp tables differ.
+        """
+        if self.data is None:
+            raise RuntimeError("No table data was found to convert to a transmission curve!")
+
+        throughputKey = None
+        if 'wavelength' not in self.data.columns:
+            raise RuntimeError("Expected column [wavelength] not found.")
+        if 'efficiency' in self.data.columns:
+            throughputKey = 'efficiency'
+        elif 'throughput' in self.data.columns:
+            throughputKey = 'throughput'
+        else:
+            raise RuntimeError("Expected columns [throughput|efficiency] not found.")
+
+        doAverageCurves = False
+        if 'amp_name' in self.data.columns:
+            doAverageCurves = True
+
+        # These will be used to construct the
+        # ``lsst.afw.image.TransmissionCurve`` object.
+        wavelengths = None
+        throughput = None
+
+        if doAverageCurves:
+            curveStack = None
+            amplifierNames = set(self.data['amp_name'])
+            comparisonIndices = np.where(self.data['amp_name'] == next(iter(amplifierNames)))
+            wavelengths = self.data[comparisonIndices]['wavelength']
+
+            for amplifier in amplifierNames:
+                indices = np.where(self.data['amp_name'] == amplifier)
+                if len(self.data[indices]) != len(self.data[comparisonIndices]):
+                    raise RuntimeError("Incompatible lengths in average.")
+                if not np.array_equal(self.data[indices]['wavelength'], wavelengths):
+                    raise RuntimeError("Mismatch in wavelength samples.")
+
+                if curveStack is not None:
+                    curveStack = np.column_stack((curveStack, self.data[indices][throughputKey]))
+                else:
+                    curveStack = self.data[indices][throughputKey]
+            throughput = np.mean(curveStack, 1)
+
+            # This averaging operation has stripped units.
+            throughput = throughput * self.data[throughputKey].unit
+        else:
+            wavelengths = self.data['wavelength']
+            throughput = self.data[throughputKey]
+
+        # Convert units:
+        # import pdb; pdb.set_trace()
+        with cds.enable():
+            # These need to be in Angstroms, for consistency.
+            wavelengths = wavelengths.to(u.Angstrom).to_value()
+            # This is ugly.  Fix.
```

Comment:
Does 'fix' mean 'to be fixed' or is the fix below?

Reply:
It remains ugly (to me at least), but the subsequent `if` line is the fix.  I have deleted this comment as it was intended as an intermediate note to myself.